packages/round-manager/src/context/ApplicationContext.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { createContext, useContext, useEffect, useReducer } from "react";
+import { GrantApplication } from "../features/api/types";
+import { useWallet } from "../features/common/Auth";
+import { getApplicationById } from "../features/api/application";
+
+enum ActionType {
+  SET_APPLICATIONS = "SET_APPLICATIONS",
+  SET_LOADING = "SET_LOADING",
+  SET_ERROR_GET_APPLICATION = "SET_ERROR_GET_APPLICATION",
+}
+
+interface Action {
+  type: ActionType;
+  payload?: any;
+}
+
+type Dispatch = (action: Action) => void;
+
+export interface ApplicationState {
+  application?: GrantApplication;
+  isLoading: boolean;
+  getApplicationByIdError?: Error;
+}
+
+export const ApplicationContext = createContext<
+  { state: ApplicationState; dispatch: Dispatch } | undefined
+>(undefined);
+
+const applicationReducer = (
+  state: ApplicationState,
+  action: Action
+): ApplicationState => {
+  switch (action.type) {
+    case ActionType.SET_APPLICATIONS:
+      return {
+        ...state,
+        application: action.payload,
+      };
+    case ActionType.SET_LOADING:
+      return {
+        ...state,
+        isLoading: action.payload,
+      };
+    case ActionType.SET_ERROR_GET_APPLICATION:
+      return {
+        ...state,
+        getApplicationByIdError: action.payload,
+      };
+  }
+
+  return state;
+};
+
+export const initialApplicationState: ApplicationState = {
+  application: undefined,
+  isLoading: false,
+};
+
+export const ApplicationProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [state, dispatch] = useReducer(
+    applicationReducer,
+    initialApplicationState
+  );
+
+  const providerProps = {
+    state,
+    dispatch,
+  };
+
+  return (
+    <ApplicationContext.Provider value={providerProps}>
+      {children}
+    </ApplicationContext.Provider>
+  );
+};
+
+function fetchApplicationById(
+  dispatch: Dispatch,
+  id: string | undefined,
```

Comment:
imo we should handle potential undefined `id` in `useApplicationById` - and `fetchApplicationById` shouldn't even be called if `id` is undefined